klippy/extras/gcode_move.py

```diff
@@ -102,35 +105,47 @@ def get_status(self, eventtime=None):
             'extrude_factor': self.extrude_factor,
             'absolute_coordinates': self.absolute_coord,
             'absolute_extrude': self.absolute_extrude,
-            'homing_origin': self.Coord(*self.homing_position),
-            'position': self.Coord(*self.last_position),
-            'gcode_position': self.Coord(*move_position),
+            'homing_origin': self.Coord(*self.homing_position[:4]),
+            'position': self.Coord(*self.last_position[:4]),
+            'gcode_position': self.Coord(*move_position[:4]),
         }
     def reset_last_position(self):
         if self.is_printer_ready:
             self.last_position = self.position_with_transform()
+    def _update_extra_axes(self):
+        toolhead = self.printer.lookup_object('toolhead')
+        axis_map = {'X':0, 'Y': 1, 'Z': 2, 'E': 3}
+        extra_axes = toolhead.get_extra_axes()
+        for index, ea in enumerate(extra_axes):
+            if ea is None:
+                continue
+            gcode_id = ea.get_axis_gcode_id()
+            if gcode_id is None or gcode_id in axis_map or gcode_id in "FN":
+                continue
+            axis_map[gcode_id] = index
```

Comment:
I understand this index needs an offset to work correctly in cmd_G1 below.

Reply:
The code looks okay to me.  Maybe I'm missing something.  The `toolhead.get_extra_axes()` code always returns a list that matches the fields needed by `toolhead.move()`.  So, if index 5 is a manual_stepper in `get_extra_axes()` then index 5 in `position` sent to `move()` should be the position of that manual_stepper.